### PR TITLE
Set the relationship type

### PR DESCRIPTION
**Pull request by melohagan**
## Description
The user field component makes use of the relationship field component, therefore we need to ensure the relationshipType is set.
Undefined defaults to multi-select.

Addresses:
- https://github.com/Budibase/budibase/issues/12272
- https://github.com/Budibase/budibase/issues/12302

## Screenshots

Screenshot 2023-11-03 at 15 07 27

Screenshot 2023-11-03 at 15 07 47

rel_bug

### CODE REVIEW

File: packages/builder/src/components/backend/DataTable/modals/CreateEditColumn.svelte [hunk 1]
```diff
@@ -143,6 +144,8 @@
       )?.[0]
       // Set the tableId based on the selected table
       editableColumn.tableId = relationshipTableIdSecondary
+    } else if (editableColumn.subtype === USER_TYPE) {
```
Comment (adrinr) [inline packages/builder/src/components/backend/DataTable/modals/CreateEditColumn.svelte:147]: Don't we need the `USERS` type as well?

Comment (melohagan) [reply packages/builder/src/components/backend/DataTable/modals/CreateEditColumn.svelte:147]: The picker does default to many, but it probably is worth being explicit.

Would multiple users be `many-to-one` or `many-to-many` in this case?

Comment (melohagan) [reply packages/builder/src/components/backend/DataTable/modals/CreateEditColumn.svelte:148]: users type added

File: packages/builder/src/components/backend/DataTable/modals/CreateEditColumn.svelte [hunk 2]
```diff
@@ -143,6 +145,10 @@
...
+      editableColumn.relationshipType = RelationshipType.ONE_TO_MANY
+    } else if (editableColumn.subtype === USERS_TYPE) {
+      editableColumn.relationshipType = RelationshipType.MANY_TO_MANY
```
Comment (mike12345567) [inline packages/builder/src/components/backend/DataTable/modals/CreateEditColumn.svelte:151]: Can you update the `BBReferenceFieldMetadata` type in `types/src/documents/app/table/schema.ts` to include the `relationshipType` field as the `RelationshipType` enum - just for future reference, while it isn't used on backend, good to know the structures used.

**Review by mike12345567 — CHANGES_REQUESTED** [review]
One small change - once thats done good to go!

File: packages/builder/src/components/backend/DataTable/modals/CreateEditColumn.svelte [hunk 3]
```diff
@@ -287,6 +289,12 @@
     if (saveColumn.type !== LINK_TYPE) {
       delete saveColumn.fieldName
     }
+    if (saveColumn.subtype === USER_TYPE) {
```
Comment (mike12345567) [inline packages/builder/src/components/backend/DataTable/modals/CreateEditColumn.svelte:292]: should you confirm it is a bbReference type first just to make sure that it is a user column?

Comment (melohagan) [reply packages/builder/src/components/backend/DataTable/modals/CreateEditColumn.svelte]: Now checking bb_reference type

**Review by mike12345567 — APPROVED** [review]
LGTM!

**Pull request merged**